### PR TITLE
Add InfraDiff class for feast plan

### PR DESCRIPTION
**Pull request by felixwang9817**
**What this PR does / why we need it**: This PR adds the `InfraDiff` class and adds support for `Infra` objects to the `Registry`.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/diff/utils.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from dataclasses import dataclass
+from enum import Enum
```
Comment (achals) [inline sdk/python/feast/diff/utils.py:2]: nit, prefer a different name for this file than `utils.py`? It's a smell

**Review by achals — COMMENTED** [review]
Can you update the commit message/PR title should be more descriptive

**Review by achals — APPROVED** [review]
Can you also print out what `feast plan` produces when infra is changed in the PR description?

**Comment by felixwang9817** [comment]
> Can you also print out what `feast plan` produces when infra is changed in the PR description?

this PR doesn't actually change the output of `feast plan`

**Pull request merged**